tools/cldr-code/src/test/java/org/unicode/cldr/test/TestEnInheritance.java

```diff
@@ -0,0 +1,46 @@
+package org.unicode.cldr.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.logging.Logger;
+import org.junit.jupiter.api.Test;
+import org.unicode.cldr.util.*;
+
+/**
+ * Check modern coverage inherited values in en that are marked with up arrows or are blank; all
+ * values should be explicit in en
+ */
+public class TestEnInheritance {
+    private static final Logger logger = Logger.getLogger(TestEnInheritance.class.getName());
+
+    private static final String LOCALE_ID = "en";
+
+    @Test
+    void testEn() {
+        final CLDRConfig config = CLDRConfig.getInstance();
+        final CLDRFile cldrFile = config.getCLDRFile(LOCALE_ID, true);
+        final SupplementalDataInfo sdi = SupplementalDataInfo.getInstance();
+        final CoverageLevel2 coverageLevel = CoverageLevel2.getInstance(sdi, LOCALE_ID);
+        int pathsWithNullValue = 0, pathsWithBlankValue = 0, pathsWithMarkerValue = 0;
+        for (final String path : cldrFile.fullIterable()) {
+            if (coverageLevel.getLevel(path).getLevel() <= Level.MODERN.getLevel()) {
+                String value = cldrFile.getStringValue(path);
+                if (value == null) {
+                    complain("null value", ++pathsWithNullValue, path);
+                } else if (value.isBlank()) {
```

Comment:
There are a few values that can be blank, so they would need to be excluded. I can't remember if these are marked in the DTD or not; if not, we might consider that so that they can be tested with DtdData.

Reply:
nativeSpaceReplacement and foreignSpaceReplacement and ...?

Reply:
The second commit adds exclusions for nativeSpaceReplacement and foreignSpaceReplacement based on their constants DisplayAndInputProcessor.FSR_START_PATH/NSR_START_PATH -- not sure how we'd use the DTD...?